Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs

```diff
@@ -47,24 +47,26 @@ public override void OnReload(GunStats stats)
     public override void OnFire(GunStats stats)
     {
         animator.SetTrigger("Fire");
-        for (int i = stats.Ammo; i < magazineSize; i++)
+        for (int i = stats.Ammo - 1; i < magazineSize; i++)
```

Comment:
I noticed that when you've got 1 ammo left in the hat barrel, there are still two hats left:  
![image](https://github.com/hackerspace-ntnu/Red-Planet-Rampage/assets/44986729/95a1ee76-8fe7-49e8-bb77-d586d172a1ba)
I believe this could be fixed by subtracting **2** from the ammo count in this for loop, and preventing -1 indexing.

Reply:
I had to do some additional state management, but yeah is fixed now 😅